src/owner.rs

```diff
@@ -0,0 +1,325 @@
+                              https://crates.io/login"));
+        }
+        _ => {
+            return Err(internal(format!("didn't get a 200 result from github: {}",
+                                    resp)))
+        }
+    }
+
+    #[derive(RustcDecodable)]
+    struct Membership {
+        state: String,
+    }
+    let json = try!(str::from_utf8(resp.get_body()).ok().chain_error(||{
+        internal("github didn't send a utf8-response")
+    }));
+    let membership: Membership = try!(json::decode(json).chain_error(|| {
+        internal("github didn't send a valid json response")
+    }));
+
+    // There is also `state: pending` for which we could possibly give
+    // some feedback, but it's not obvious how that should work.
+    Ok(membership.state == "active")
+}
+
+impl Model for Team {
+    fn from_row(row: &Row) -> Self {
+        Team {
+            id: row.get("id"),
+            name: row.get("name"),
+            github_id: row.get("github_id"),
+        }
+    }
+
+    fn table_name(_: Option<Self>) -> &'static str { "teams" }
+}
+
+impl Owner {
+    /// Finds the owner by name, failing out if it doesn't exist.
+    /// May be a user's GH login, or a full team name. This is case
+    /// sensitive.
+    pub fn find_by_name(conn: &Connection, name: &str) -> CargoResult<Owner> {
+        let owner = if name.contains(":") {
+            Owner::Team(try!(Team::find_by_name(conn, name).map_err(|_|
+                human(format!("could not find team with name {}", name))
+            )))
+        } else {
+            Owner::User(try!(User::find_by_login(conn, name).map_err(|_|
+                human(format!("could not find user with login `{}`", name))
+            )))
+        };
+        Ok(owner)
+    }
+
+    pub fn kind(&self) -> i32 {
+        match *self {
+            Owner::User(_) => OwnerKind::User as i32,
+            Owner::Team(_) => OwnerKind::Team as i32,
+        }
+    }
+
+    pub fn name(&self) -> &str {
+        match *self {
+            Owner::User(ref user) => &user.gh_login,
+            Owner::Team(ref team) => &team.name,
+        }
+    }
+
+    pub fn id(&self) -> i32 {
+        match *self {
+            Owner::User(ref user) => user.id,
+            Owner::Team(ref team) => team.id,
+        }
+    }
+
+    pub fn encodable(self) -> EncodableOwner {
+        match self {
+            Owner::User(User { id, email, name, gh_login, avatar, .. }) => {
+                EncodableOwner {
+                    id: id,
+                    login: gh_login,
+                    email: email,
+                    avatar: avatar,
+                    name: name,
+                    kind: String::from("user"),
+                }
+            }
+            Owner::Team(Team { id, name, .. }) => {
+                EncodableOwner {
+                    id: id,
+                    login: name,
+                    email: None,
+                    avatar: None,
```

Comment:
The frontend really drops the ball on this, I'm not gonna lie. Broken image and a link to `github.com/github:rust-lang:owners`